include/tvm/runtime/vm.h

```diff
@@ -25,34 +25,48 @@
 #define TVM_RUNTIME_VM_H_
 
 #include <tvm/runtime/object.h>
+#include <tvm/runtime/memory.h>
 #include <tvm/runtime/packed_func.h>
 #include <tvm/runtime/registry.h>
 #include <memory>
 #include <string>
 #include <unordered_map>
+#include <utility>
 #include <vector>
 
 namespace tvm {
 namespace runtime {
 namespace vm {
 
-/*! \brief An object representing a closure. */
+/*!
+ * \brief An object representing a closure. This object is used by both the
+ * Relay VM and interpreter.
+ */
 class ClosureObj : public Object {
  public:
-  /*! \brief The index into the VM function table. */
+  /*!
+   * \brief The index into the function list. The function could be any
+   * function object that is compatible to a certain runtime, i.e. VM or
+   * interpreter.
+   */
   size_t func_index;
   /*! \brief The free variables of the closure. */
   std::vector<ObjectRef> free_vars;
 
-  static constexpr const uint32_t _type_index = TypeIndex::kVMClosure;
-  static constexpr const char* _type_key = "vm.Closure";
```

Comment:
@tqchen and I discussed, we feel like it is probably better to keep the keys namespaced, and eventually move them to the `runtime` namespace?

Reply:
Sure. I will change them back.